src/app/views/authentication/Authentication.tsx

```diff
@@ -1,182 +1,152 @@
-          const authResponse = hello('msft').getAuthResponse();
-          accessToken = authResponse.access_token;
-          if (accessToken) {
-            authenticated.token = accessToken;
-          }
-          authenticated.status = true;
-          if (actions) {
-            actions.authenticateUser(authenticated);
-          }
-        }
-        if (accessToken) {
-          try {
-            const query = {
-              sampleURL : USER_INFO_URL,
-            };
-            const userInfo = (queryActions) ? await queryActions.runQuery(query) : null;
-            const jsonUserInfo = userInfo.response.body;
-            authenticated.user = {...{},
-                                  displayName: jsonUserInfo.displayName,
-                                  emailAddress: jsonUserInfo.mail || jsonUserInfo.userPrincipalName,
-                                  profileImageUrl: '',
-            };
-            if (actions) {
-              actions.authenticateUser(authenticated);
-              this.setState({
-                authenticated,
-              });
-            }
-            try {
-              const pictureQuery = {
-                sampleURL : USER_PICTURE_URL,
-              };
-              const userPicture = (queryActions) ? await queryActions.runQuery(pictureQuery) : null;
-              const buffer = await userPicture.response.body.arrayBuffer();
-              const blob = new Blob([buffer], { type: 'image/jpeg' });
-              const imageUrl = URL.createObjectURL(blob);
-              if (actions) {
-                authenticated = this.state.authenticated;
-                authenticated.user.profileImageUrl = imageUrl;
-                actions.authenticateUser(authenticated);
-                this.setState({
-                  authenticated,
-                  loading: false,
-                });
-              }
-            } catch (e) {
-              if (actions) {
-                authenticated = this.state.authenticated;
-                authenticated.user.profileImageUrl = '';
-                actions.authenticateUser(authenticated);
-                this.setState({
-                  authenticated,
-                  loading: false,
-                });
-              }
-            }
-          } catch (e) {
-            // tslint:disable-next-line:no-console
-            console.log(e);
-          }
-        }
-      });
+  private async getUserInfo(queryActions: any) {
+    const userInfo = (queryActions) ? await queryActions.runQuery({
+      sampleURL: USER_INFO_URL,
+    }) : null;
+    const jsonUserInfo = userInfo.response.body;
+    return jsonUserInfo;
+  }
+
+  private async getImageUrl(queryActions: any) {
+    const userPicture = (queryActions) ? await queryActions.runQuery({
+      sampleURL: USER_PICTURE_URL,
+    }) : null;
+    const buffer = await userPicture.response.body.arrayBuffer();
+    const blob = new Blob([buffer], { type: 'image/jpeg' });
```

Comment:
Have we tested that type "image/jpeg" would work for all types of images ?

Reply:
We carried this over from classic. As there were no noted problems with the image before, its highly unlikely for there to be a problem now.